cmake/CMakeLists.txt

```diff
@@ -238,6 +254,14 @@ else()
     string(APPEND CMAKE_C_FLAGS_RELEASE " -march=native -mtune=native")
     string(APPEND CMAKE_CXX_FLAGS_RELWITHDEBINFO " -march=native -mtune=native")
     string(APPEND CMAKE_C_FLAGS_RELWITHDEBINFO " -march=native -mtune=native")
+  elseif  (onnxruntime_USE_AVX)
+    SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mavx")
+  elseif  (onnxruntime_USE_AVX2)
+    SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mavx2")
+  elseif  (onnxruntime_USE_AVX512)
+    SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mavx512f  -mavx512pf  -mavx512er  -mavx512cd  -mavx512vl"
+            " -mavx512bw  -mavx512dq  -mavx512ifma  -mavx512vbmi -mavx512vbmi2 -mavx512vpopcntdq -mavx5124fmaps"
```

Comment:
This is enabling many instruction set extensions that aren't common place. I would stick with -mavx512f, -mavx512cd, -mavx512bw, -mavx512dq, and -mavx512vl. The switches -mavx5124maps, -mavx512pf, and -mavx512er are for old Knights Mill and Knights Landing CPUs that aren't readily available today. The others are for new instructions that shouldn't be turned on automatically.